backend/server/routers/auth.py

```diff
@@ -1,14 +1,276 @@
 """ Routes to deal with user Authentication. """
-from fastapi import APIRouter
+from typing import Annotated, Dict, Optional, Tuple, Union, cast
+from datetime import datetime
+from secrets import token_hex, token_urlsafe
+from time import time
+from fastapi import APIRouter, Cookie, HTTPException, Response, Security
+from pydantic import BaseModel
+from starlette.status import HTTP_401_UNAUTHORIZED, HTTP_400_BAD_REQUEST, HTTP_500_INTERNAL_SERVER_ERROR
 
-from server.routers.user import default_cs_user, reset, set_user
+from server.config import SECURE_COOKIES
+from server.db.helpers.models import NotSetupUserStorage, GuestSessionInfoModel, RefreshToken, SessionID, SessionInfoModel, SessionOIDCInfoModel, SessionToken
+from server.db.helpers.users import delete_user, insert_new_user
+
+from .auth_utility.sessions.errors import ExpiredRefreshTokenError, ExpiredSessionTokenError, OldRefreshTokenError
+from .auth_utility.sessions.interface import create_new_guest_token_pair, get_session_info_from_refresh_token, get_session_info_from_session_token, logout_session, setup_new_csesoc_session, create_new_csesoc_token_pair, setup_new_guest_session
+
+from .auth_utility.middleware import HTTPBearer401, set_secure_cookie
+from .auth_utility.oidc.requests import DecodedIDToken, exchange_and_validate, generate_oidc_auth_url, get_userinfo_and_validate, refresh_and_validate, revoke_token, validate_authorization_response
+from .auth_utility.oidc.errors import OIDCInvalidGrant, OIDCInvalidToken, OIDCTokenError, OIDCValidationError
+
+STATE_TTL = 10 * 60
```

Comment:
We should be more specific about what this it a TTL for

Reply:
really true, i will rename it to `AUTH_STATE_COOKIE_TTL` to match the other constants,